### PR TITLE
8.3.x pantheon

### PR DESCRIPTION
**Pull request by alexellison**
@tauno @mariacha would appreciate if you gave this a quick once-over.

Main changes are upgrade to Drupal 8.3.x and a few necessary config changes to make install go smoothly after this upgrade.

I also added several new modules to our baseline (including thinkshout/media) and updated the versions of all relevant contrib modules.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,40 +14,57 @@
...
-        "drupal/migrate_plus": "3.0.0-beta1",
-        "drupal/pathauto": "1.0.0-beta1",
-        "drupal/simple_sitemap": "2.7.0",
-        "drupal/token": "1.0.0-beta2",
-        "drupal/config_installer": "1.3.0"
+        "drupal/admin_toolbar": "^1.18",
+        "drupal/config_installer": "^1.3",
+        "drupal/contact_storage": "^1.0-beta8",
+        "drupal/ctools": "^3.0-alpha27",
+        "drupal/devel": "^1.0-rc1",
```
Comment (mariacha) [inline composer.json]: Should we move devel to the require-dev area?

File: config/core.extension.yml [hunk 2]
```diff
@@ -46,5 +46,6 @@ theme:
   classy: 0
   bartik: 0
   seven: 0
+profile: config_installer
```
Comment (mariacha) [inline config/core.extension.yml:49]: Is this added by 8.3?

Comment (mariacha) [reply config/core.extension.yml:49]: @tauno You mentioned once that config_installer isn't really a profile. Could we say something like "standard" or even "lightning" here and still have the flow work?

File: composer.json (hunk 1 above)
Comment (alexellison) [reply composer.json]: Yes; good idea.

File: config/core.extension.yml (hunk 2 above)
Comment (alexellison) [reply config/core.extension.yml:49]: This is required in 8.3 due to changes made in [40f803a](https://www.drupal.org/commitlog/commit/2/40f803ac9dd3e36926d5236e74e01e48a03fa51a) specifically, [this line](http://cgit.drupalcode.org/drupal/tree/core/lib/Drupal/Core/EventSubscriber/ConfigImportSubscriber.php?id=40f803ac9dd3e36926d5236e74e01e48a03fa51a#n138) requires a profile to be set so it's guaranteed the post-install profile matches the pre-install (config) profile.

For this reason we can't use "standard" or anything other than "config_installer".

**Comment by mariacha** [comment]
This PR also has a couple low-risk things that might be nice to pull in with this set of changes:
https://github.com/thinkshout/drupal-project/pull/12/files

.gitignore for Theming
move .htaccess from the root to the /web folder

**Comment by alexellison** [comment]
@mariacha I'm for merging #12 with #15 before we merge both back into `8.x-pantheon`.
I'll try to make time for a review of #12 soon.

**Comment by mariacha** [comment]
Alright then. My only real feedback is that devel item then. Thanks for taking the time to do this!

**Comment by tauno** [comment]
Alex - the config_installer profile used to set the install profile to
minimal (or standard) after installing. This was necessary because the
config_installer didn't remain enabled.

**Comment by alexellison** [comment]
Got it - @tauno. config_installer now sets the install profile to `config_installer` in `settings.php` after installing, which I accounted for in [6189f11](https://github.com/thinkshout/drupal-project/commit/6189f11b237db47c36b129f8ae226aab970caff7).

**Comment by mariacha** [comment]
I'm impatient and am merging this.

**Pull request merged**